pkg/querier/blocks_store_queryable.go

```diff
@@ -513,6 +513,11 @@ func (q *blocksStoreQuerier) queryWithConsistencyCheck(ctx context.Context, logg
 
 	// Find the list of blocks we need to query given the time range.
 	knownBlocks, knownDeletionMarks, err := q.finder.GetBlocks(ctx, userID, minT, maxT)
```

Comment:
There is no way for us to avoid finding blocks again because we need to get deletion marks?

Reply:
Maybe, but I noticed both finder implementations cache this call, so I didn’t bother.
What do you think?

Reply:
I am ok with it